core-contracts/Dex/src/main/java/network/balanced/score/core/dex/DexImpl.java

```diff
@@ -0,0 +1,1544 @@
+
+        if (value.compareTo(BigInteger.ZERO) > 0) {
+            activeAddresses.get(id).add(to);
+        }
+        if (balance.at(id).getOrDefault(from, BigInteger.ZERO).compareTo(BigInteger.ZERO) == 0) {
+            activeAddresses.get(id).remove(from);
+        }
+        TransferSingle(from, from, to, id, value);
+
+        updateAccountSnapshot(from, id);
+        updateAccountSnapshot(to, id);
+
+        if (to.isContract()) {
+            Context.call(to, "onIRC31Received", from, from, id, value, data);
+        }
+    }
+
+    @External
+    public void onIRC31Received(Address _operator, Address _from,
+                                BigInteger _id, BigInteger _value,
+                                byte[] _data) {
+        Context.revert(TAG + ": IRC31 Tokens not accepted");
+
+    }
+
+    @External
+    public boolean precompute(BigInteger snap, BigInteger batch_size) {
+        return true;
+    }
+
+    @External(readonly = true)
+    public BigInteger getDeposit(Address _tokenAddress, Address _user) {
+        return deposit.at(_tokenAddress).getOrDefault(_user, BigInteger.ZERO);
+    }
+
+    @External(readonly = true)
+    public BigInteger getSicxEarnings(Address _user) {
+        return sicxEarnings.getOrDefault(_user, BigInteger.ZERO);
+    }
+
+    @External(readonly = true)
+    public BigInteger getWithdrawLock(BigInteger _id, Address _owner) {
+        return withdrawLock.at(_id).get(_owner);
+    }
+
+    @External(readonly = true)
+    public BigInteger getPoolId(Address _token1Address, Address _token2Address) {
+        return poolId.at(_token1Address).get(_token2Address);
+    }
+
+    @External(readonly = true)
+    public BigInteger getNonce() {
+        return nonce.getOrDefault(BigInteger.ZERO);
+    }
+
+    @External(readonly = true)
+    public List<String> getNamedPools() {
+        List<String> namedPools = new ArrayList<>();
+        namedPools.addAll(namedMarkets.keys());
+        return namedPools;
+    }
+
+    @External(readonly = true)
+    public BigInteger lookupPid(String _name) {
+        return namedMarkets.get(_name);
+    }
+
+    @External(readonly = true)
+    public BigInteger getPoolTotal(BigInteger _id, Address _token) {
+        return poolTotal.at(_id).get(_token);
+    }
+
+    @External(readonly = true)
+    public BigInteger totalSupply(BigInteger _id) {
+        if (_id.equals(SICXICX_POOL_ID)) {
+            return icxQueueTotal.getOrDefault(BigInteger.ZERO);
+        }
+        return total.getOrDefault(_id, BigInteger.ZERO);
+    }
+
+    @External(readonly = true)
+    public BigInteger balanceOf(Address _owner, BigInteger _id) {
+        if (_id.equals(SICXICX_POOL_ID)) {
+            BigInteger orderId = icxQueueOrderId.get(_owner);
+            if (orderId == null) {
+                return BigInteger.ZERO;
+            }
+            return icxQueue.getNode(orderId).getSize();
+        } else {
+            return balance.at(_id).get(_owner);
+        }
```

Comment:
We need something like this to take into account stakedLP balance before migration
```suggestion
        } else {
            BigInteger balance = this.balance.at(_id).getOrDefault(_owner, BigInteger.ZERO);
            if (getDay().compareTo(continuousRewardsDay.get()) <= 0) {
                BigInteger stakedBalance = (BigInteger) Context.call(stakedlp.get(), "balanceOf", _owner, poolId);
                balance = balance.add(stakedBalance);
            }
            
            return balance;
    }
```

Reply:
Investigate that what changes are needed for sicx/icx pool